op-bindings/bindings/registry.go

```diff
@@ -19,11 +24,36 @@ func GetStorageLayout(name string) (*solc.StorageLayout, error) {
 	return layout, nil
 }
 
+// GetDeployedBytecode returns the deployed bytecode of a contract by name.
 func GetDeployedBytecode(name string) ([]byte, error) {
 	bc := deployedBytecodes[name]
 	if bc == "" {
 		return nil, fmt.Errorf("%s: deployed bytecode not found", name)
 	}
 
+	if !isHex(bc) {
+		return nil, fmt.Errorf("%s: invalid deployed bytecode", name)
+	}
+
 	return common.FromHex(bc), nil
 }
+
+// isHexCharacter returns bool of c being a valid hexadecimal.
+func isHexCharacter(c byte) bool {
+	return ('0' <= c && c <= '9') || ('a' <= c && c <= 'f') || ('A' <= c && c <= 'F')
+}
+
+// isHex validates whether each byte is valid hexadecimal string.
+func isHex(str string) bool {
+	if len(str)%2 != 0 {
+		return false
+	}
+	str = strings.TrimPrefix(str, "0x")
+
+	for _, c := range []byte(str) {
+		if !isHexCharacter(c) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
I think we should be able to use the builtin golang `encoding/hex` library here to avoid regex typos. See: https://pkg.go.dev/encoding/hex#DecodeString

For example, we could have isHex just check if `hex.DecodeString(s)` returns an error. Something like:
```golang
// isHex validates whether each byte is valid hexadecimal string.
func isHex(str string) bool {
	if len(str)%2 != 0 {
		return false
	}
	str = strings.TrimPrefix(str, "0x")

    _, err := hex.DecodeString(str)
    if err != nil {
         return false
    }
    return true
}
```

Reply:
The above code was used in go-ethereum so I'm not worried about it having a bug